Makefile

```diff
@@ -43,10 +43,12 @@ docs: ## build the docs docker image
 setup-dev-env:
 	$(MAKE) dev
 
+.PHONY: lint
+lint:
+	@golangci-lint run ./...
```

Comment:
Can we not do everything through makefile so that the user's golangci version and gha's golangci's version is the same?
We will have to do go install though.
Perhaps we can re-use docker instruction to make it platform agonstic and independent of the users environment.
https://golangci-lint.run/usage/install/

Reply:
> We will have to do go install though.

You cannot Go install golanci-lint, it does not work. And while I am good with providing a docker version, golangci-lint uses a cache to make it fast. Just running it in docker will be very slow and painful. The Github action allows this caching in the CI, and running locally uses the cache, which will change position between systems.

TLDR; You can, but I would not suggest it.

Reply:
I will leave upto you to decide if you want to use Docker or not. 

However, I think we should consider that this file moves to ci.yml and have the Makefile rule invoke linting. There are some benefits:
1. Makefile will make it consistent between Github actions and local environment.
2. Common ci.yaml will help it easier to control go version and golangci-lint versions as well.

I am not convinced above caching and speed since other CI jobs will definitely take longer than golangci-lint.